### PR TITLE
Hs/userstore db

### PR DESCRIPTION
**Pull request by Half-Shot**
This depends on #383.

Currently missing tests and fixes to the tools.

### CODE REVIEW

File: src/db/userstore.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        if (!row) {
+            return null;
+        }
+        const remoteUser = new RemoteUser(remoteId);
+        const data = await this.db.Get(
+            "SELECT * FROM remote_user_data WHERE remote_id = $remoteId",
+            {remoteId},
+        );
+        if (data) {
+            remoteUser.avatarurl = data.avatarurl as string|null;
```
Comment (Sorunome) [inline src/db/userstore.ts]: Maybe also return `null` if no remote data is found? Same argumentation as with channel store

File: src/db/userstore.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                remote_id: user.id,
+            });
+        }
+
+    }
+
+    public async linkUsers(matrixId: string, remoteId: string) {
+        // This is used  ONCE in the bridge to link two IDs, so do not UPSURT data.
+        try {
+            await this.db.Run(`INSERT INTO user_entries VALUES ($matrixId, $remoteId)`, {
```
Comment (Sorunome) [inline src/db/userstore.ts]: check if that key-value pair already exists?

File: src/discordas.ts [hunk 3]
```diff
@@ -149,11 +149,16 @@ async function run(port: number, fileConfig: DiscordBridgeConfig) {
...
+        log.warn("[DEPRECATED] The user store is now part of the SQL database."
+               + "The config option userStorePath no longer has any use.");
+    }
+
     await bridge.run(port, config);
     log.info(`Started listening on port ${port}`);
 
     try {
-        await store.init(undefined, bridge.getRoomStore());
+        await store.init(undefined, bridge.getRoomStore(), bridge.getUserStore());
```
Comment (Sorunome) [inline src/discordas.ts]: shouldn't store.init stuff go into bridge.init ?

**Comment by turt2live** [comment]
@Half-Shot This shows up in the logs a lot after a 200 OK on updating the profile for a user:

```
Feb-16 07:00:59.448 [UserSync] error: Failed to update user's profile { error: syntax error at or near "WHERE"
    at Connection.parseE (/node_modules/pg/lib/connection.js:554:11)
    at Connection.parseMessage (/node_modules/pg/lib/connection.js:379:19)
    at Socket.<anonymous> (/node_modules/pg/lib/connection.js:119:22)
    at Socket.emit (events.js:197:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)
  name: 'error',
  length: 88,
  severity: 'ERROR',
  code: '42601',
  detail: undefined,
  hint: undefined,
  position: '243',
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'scan.l',
  line: '1124',
  routine: 'scanner_yyerror' }
```

**Comment by Half-Shot** [comment]
Ta

**Comment by Sorunome** [comment]
@Half-Shot was there anything still missing with this?

**Comment by Half-Shot** [comment]
Probably fixing the issue that @turt2live posted, otherwise it's good.

**Pull request merged**